### PR TITLE
SoundFX/PaintingPuzzle

### PR DESCRIPTION
**Pull request by HugoLandin**
SoundFX in assets, PaintingPuzzle in prefabs

### CODE REVIEW

File: Assets/Painting.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    private void Awake()
+    {
+        paintingRenderer = GetComponentInChildren<MeshRenderer>();
+        light = GetComponentInChildren<Light>();
+        paintingManager = GetComponentInParent<PaintingManager>();
+        OnInteract += SetSelected;
+    }
+    private void Update()
+    {
+        light.range = (Convert.ToInt32(paintingManager.selectedPainting == this)*3);
```
Comment (trilleplay) [inline Assets/Painting.cs:20]: wtf

**Review by trilleplay — CHANGES_REQUESTED** [review]

File: Assets/Painting.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Painting : Interactable
+{
+    PaintingManager paintingManager;
```
Comment (olchyk98) [inline Assets/Painting.cs:8]: private

File: Assets/Painting.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    public Light light;
```
Comment (olchyk98) [inline Assets/Painting.cs:9]: uh?

File: Assets/Painting.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    public MeshRenderer paintingRenderer;
```
Comment (olchyk98) [inline Assets/Painting.cs:10]: should it be pub?
If the answer is yes then PaintingRenderer

File: Assets/Painting.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    private void Awake()
```
Comment (olchyk98) [inline Assets/Painting.cs:11]: start?

File: Assets/PaintingManager.cs [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+using UnityEngine;
+using UnityEngine.Events;
+
+public class PaintingManager : Interactable
+{
+    public Painting selectedPainting;
+    public UnityEvent CompleteEvent;
+    System.Random rNG = new System.Random();
+    [SerializeField]
+    Painting[] paintings;
```
Comment (olchyk98) [inline Assets/PaintingManager.cs:12]: private
_paintings

**Review by olchyk98 — CHANGES_REQUESTED** [review]

File: Assets/PaintingManager.cs [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    public List<Material> fakePaintings;
+    public Material realPainting;
+    int correctPaintingIndex;
```
[the remark this answers is not in the document]
Comment (HugoLandin) [reply Assets/PaintingManager.cs:15]: k0ded will solve that later

File: Assets/Painting.cs (hunk 4 above)
Comment (olchyk98) [reply Assets/Painting.cs:10]: ok.
but rename
all public fields should start with a capital letre

**Comment by olchyk98** [comment]
@trilleplay det kommer att ta 200 år för att rebasea. GJ

File: Assets/PaintingManager.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (olchyk98) [reply Assets/PaintingManager.cs:15]: okay, `private int correctPaintingIndex ` for now

**Comment by olchyk98** [comment]
very gud peAr i recommend

**Pull request merged**